servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/CatchAllTransportObserver.java

```diff
@@ -0,0 +1,296 @@
+/*
+ * Copyright © 2020 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.transport.netty.internal;
+
+import io.servicetalk.transport.api.ConnectionInfo;
+import io.servicetalk.transport.api.ConnectionObserver;
+import io.servicetalk.transport.api.ConnectionObserver.DataObserver;
+import io.servicetalk.transport.api.ConnectionObserver.MultiplexedObserver;
+import io.servicetalk.transport.api.ConnectionObserver.ReadObserver;
+import io.servicetalk.transport.api.ConnectionObserver.SecurityHandshakeObserver;
+import io.servicetalk.transport.api.ConnectionObserver.StreamObserver;
+import io.servicetalk.transport.api.ConnectionObserver.WriteObserver;
+import io.servicetalk.transport.api.TransportObserver;
+import io.servicetalk.transport.netty.internal.NoopTransportObserver.NoopConnectionObserver;
+import io.servicetalk.transport.netty.internal.NoopTransportObserver.NoopDataObserver;
+import io.servicetalk.transport.netty.internal.NoopTransportObserver.NoopMultiplexedObserver;
+import io.servicetalk.transport.netty.internal.NoopTransportObserver.NoopReadObserver;
+import io.servicetalk.transport.netty.internal.NoopTransportObserver.NoopSecurityHandshakeObserver;
+import io.servicetalk.transport.netty.internal.NoopTransportObserver.NoopStreamObserver;
+import io.servicetalk.transport.netty.internal.NoopTransportObserver.NoopWriteObserver;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.function.Supplier;
+import java.util.function.UnaryOperator;
+import javax.net.ssl.SSLSession;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * {@link TransportObserver} wrapper that catches and logs all exceptions.
+ */
+final class CatchAllTransportObserver implements TransportObserver {
```

Comment:
Should I make it public to be able to use in `TransportObserverConnectionFactoryFilter` with `BiTransportObserver`?